scripts/irods/test/test_configuration_reload.py

```diff
@@ -83,3 +84,37 @@ def test_how_long_restarting_takes(self):
                 IrodsController().restart(test_mode=True)
                 self.admin.assert_icommand(
                     ['irule', rule, 'null', 'ruleExecOut'], "STDOUT", [str(i)])
+
+    def test_reload_command__issue_6529(self):
+        server_config_filename = paths.server_config_path()
+        PROP_NAME = "is_issue_6529_done"
+        with lib.file_backed_up(server_config_filename):
+            # Make the configuration changes
+            with open(server_config_filename) as f:
+                svr_cfg = json.load(f)
+            svr_cfg[PROP_NAME] = "No?"
+            with open(server_config_filename, 'w') as f:
+                f.write(json.dumps(svr_cfg, sort_keys=True,
+                        indent=4, separators=(',', ': ')))
+
+            # Invoke irods-grid reload --all
+            with session.make_session_for_existing_admin() as admin_session:
+                admin_session.environment_file_contents = IrodsConfig().client_environment
```

Comment:
I'm not sure that I understand the purpose of this line. Did we modify the client environment?